api/src/main/java/org/apache/iceberg/expressions/Literals.java

```diff
@@ -501,19 +538,17 @@ public <T> Literal<T> to(Type type) {
           return (Literal<T>) new TimeLiteral(timeMicros);
 
         case TIMESTAMP:
-          if (((Types.TimestampType) type).shouldAdjustToUTC()) {
-            long timestampMicros =
-                ChronoUnit.MICROS.between(
-                    EPOCH, OffsetDateTime.parse(value(), DateTimeFormatter.ISO_DATE_TIME));
-            return (Literal<T>) new TimestampLiteral(timestampMicros);
+          final TimestampType tsType = (TimestampType) type;
+          final String value = value().toString();
+          final java.time.temporal.Temporal valueAsTemporal;
```

Comment:
Style: don't use `final` for local variables.

Reply:
Done.